### ISSUE TITLE
incorrect doc reference when methods are too close

### ISSUE THREAD

**Issue by IgnacioFan** [Shopify/ruby-lsp#2786]
While developing PR #2733, I accidentally found this very tiny bug! But there may be a different way to look at it. Anyway, just bring it up!

## Description

If we have two methods like the example, the `bar` will share the doc from `foo`. 

```ruby
module Test
  # This method shows foo
  def foo; end
  def bar; end
end

class Something
  include Test

  def foobar
    bar
  end
end
```

![Image](https://github.com/user-attachments/assets/3d04a059-becc-4aa8-9ca0-da988dd8532f)

But if we add one line between the two methods, the `bar` will act normally! 

```ruby
module Test
  # This method shows foo
  def foo
  end
  def bar; end
end

class Something
  include Test

  def foobar
    bar
  end
end
```

![Image](https://github.com/user-attachments/assets/06ce87d9-083b-4e3a-aa00-24c6514089f4)

## Additional Info

Here is where I found it over the [conversation](https://github.com/Shopify/ruby-lsp/pull/2733#discussion_r1813457346) on PR #2733

**Comment by vinistock** [comment]
Good idea, I think deleting the comments from the hash is a good strategy 👍.
